### DISCUSSION TITLE
cells.lib for $adff flip flop

### DISCUSSION THREAD

**Discussion by OuDret** [YosysHQ/yosys#4315] [Q&A] [answered]
Hello

I have made my own cells.lib but I don't know how to make my flip flops asyncron.

```
cell(DFFR) {
  ff(Q, NQ) {
          clocked_on: "C";
	  next_state: "D";
	  clear: "R";
}
  pin(D) { direction: input; }
  pin(R) { direction: input; }
  pin(C) { direction: input; clock: true; }
  pin(Q) { direction: output; function: "Q"; }
}
```
Do you have any example? Is there any documentation on this syntax?

Thank you

**Comment by nakengelhardt** [comment, marked as answer]
This is the liberty format reference: https://media.c3d2.de/mgoblin_media/media_entries/659/Liberty_User_Guides_and_Reference_Manual_Suite_Version_2017.06.pdf

E.g. `clear` is an asynchronous reset, any synchronous reset would be included in the next_state function.

**Reply by OuDret** [reply]
thank you so much
